src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,185 @@
+package duke;
+
+import java.util.ArrayList;
+
+import exceptions.DukeException;
+import exceptions.IncorrectNoOfArgumentException;
+import exceptions.InvalidCommandException;
+
+/**
+ * Represents a Parser that takes in user input and process it into the necessary format for Duke application
+ * to use.
+ *
+ * @author MrTwit99
+ * @since 2023-02-01
+ */
+public class Parser {
+
+    /**
+     * Returns an ArrayList of String type that contains the processed command information,
+     * derived from the user input via the CLI.
+     * <p></p>
+     * This method aids in processing user input.
+     *
+     * @param taskInfo Command information obtained from user's input via the CLI.
+     * @return ArrayList of String type, containing processed command information.
+     * @throws IncorrectNoOfArgumentException When there is insufficient arguments provided by the user.
+     * @see ArrayList
+     */
+    protected static ArrayList<String> parse(String taskInfo) throws IncorrectNoOfArgumentException {
```

Comment:
Very long and complicated function. Maybe you could abstract the parsing of separate parts of the string into individual functions?

Reply:
hmmm I agree with you on that as well. However, I would probably not be fixing this issue anytime soon due to the time constraint. I would look into rectifying this should time permits.